src/parser/rogue/outlaw/modules/spells/BladeFlurry.ts

```diff
@@ -0,0 +1,56 @@
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Abilities from 'parser/core/modules/Abilities';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import SPELLS from 'common/SPELLS';
+import Events, { CastEvent } from 'parser/core/Events';
+import RESOURCE_TYPES, { getResource } from 'game/RESOURCE_TYPES';
+
+class BladeFlurry extends Analyzer {
+  static dependencies = {
+    abilities: Abilities,
+    spellUsable: SpellUsable,
+  };
+
+  hasTrueBearing: boolean = false;
+  protected abilities!: Abilities;
+  protected spellUsable!: SpellUsable;
+
+  constructor(options: any) {
+    super(options);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell([SPELLS.DISPATCH, SPELLS.EVISCERATE, SPELLS.KIDNEY_SHOT, SPELLS.BETWEEN_THE_EYES, SPELLS.SLICE_AND_DICE]), this.onFinishMove);
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.TRUE_BEARING), this.onApplyTrueBearing);
+    this.addEventListener(Events.removebuff.by(SELECTED_PLAYER).spell(SPELLS.TRUE_BEARING), this.onRemoveTrueBearing);
+  }
+
+  onRemoveTrueBearing() {
```

Comment:
Are you not able to use this.selectedCombatant.hasBuff(SPELLS.TRUE_BEARING.id) instead of manually setting a variable true/false when the buff is applied/removed??

Reply:
I forgot about that function. I'll remove the unnecessary code.